pkg/controller/install/webhook.go

```diff
@@ -69,80 +68,108 @@ func (i *StrategyDeploymentInstaller) createOrUpdateWebhook(caPEM []byte, desc v
 }
 
 func (i *StrategyDeploymentInstaller) createOrUpdateMutatingWebhook(ogNamespacelabelSelector *metav1.LabelSelector, caPEM []byte, desc v1alpha1.WebhookDescription) error {
-	webhooks := []admissionregistrationv1.MutatingWebhook{
-		desc.GetMutatingWebhook(i.owner.GetNamespace(), ogNamespacelabelSelector, caPEM),
-	}
-	existingHook, err := i.strategyClient.GetOpClient().KubernetesInterface().AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), desc.Name, metav1.GetOptions{})
-	if err == nil {
-		// Check if the only owners are this CSV or in this CSV's replacement chain
-		if ownerutil.Adoptable(i.owner, existingHook.GetOwnerReferences()) {
-			ownerutil.AddNonBlockingOwner(existingHook, i.owner)
-		}
+	webhookLabels := ownerutil.OwnerLabel(i.owner, i.owner.GetObjectKind().GroupVersionKind().Kind)
+	webhookLabels[WebhookDescKey] = desc.Name
+	webhookSelector := labels.SelectorFromSet(webhookLabels).String()
 
-		// Update the list of webhooks
-		existingHook.Webhooks = webhooks
+	existingWebhooks, err := i.strategyClient.GetOpClient().KubernetesInterface().AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.TODO(), metav1.ListOptions{LabelSelector: webhookSelector})
+	if err != nil {
+		return err
+	}
 
-		// Attempt an update
-		if _, err := i.strategyClient.GetOpClient().KubernetesInterface().AdmissionregistrationV1().MutatingWebhookConfigurations().Update(context.TODO(), existingHook, metav1.UpdateOptions{}); err != nil {
-			log.Warnf("could not update MutatingWebhookConfiguration %s", existingHook.GetName())
-			return err
-		}
-	} else if k8serrors.IsNotFound(err) {
-		hook := admissionregistrationv1.MutatingWebhookConfiguration{
-			ObjectMeta: metav1.ObjectMeta{Name: desc.Name,
-				Namespace: i.owner.GetNamespace(),
+	if len(existingWebhooks.Items) == 0 {
+		// Create a MutatingWebhookConfiguration
+		webhook := admissionregistrationv1.MutatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: desc.Name + "-",
```

Comment:
I'm pretty concerned that `WebhookConfig.Name` will be confusing for a user. I would opt to add a new `WebhookConfig.GenerateName` field which follows the expected semantics. Desirable or not, as a corollary, users would still be able ensure a singleton webhooks with `WebhookConfig.Name`.

Reply:
Do we need to leave the old name? OLM hasn't cut a release since the feature was implemented.

Reply:
If that's the case then I don't seen an issue replacing `Name` w/ `GenerateName`. 

Before we go ahead, can we get some consensus among the other reviewers? If this is crazy then I don't want us to be stuck with it once we cut the next release.

Reply:
SGTM

Reply:
Completed in the latest version of the PR